terraform/modules/shared/main.tf

```diff
@@ -472,8 +472,49 @@ resource "azurerm_kubernetes_cluster" "cluster" {
   }
 
   identity {
-    type         = "UserAssigned"
-    identity_ids = [azurerm_user_assigned_identity.pipeline_runner.id]
+    type = "SystemAssigned"
+  }
+
+}
+
+data "azurerm_kubernetes_cluster" "credentials" {
+  name                = azurerm_kubernetes_cluster.cluster.name
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.azurerm_kubernetes_cluster.credentials.kube_config.0.host
+    client_certificate     = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_certificate)
+    client_key             = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_key)
+    cluster_ca_certificate = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.cluster_ca_certificate)
+  }
+}
+
+resource "helm_release" "record_linkage" {
+  name          = "phdi-${terraform.workspace}"
```

Comment:
This name ends up combining with the name of the chart (as specified in our Helm templates) to create the actual Deployment, Pod, and Service resources. In practice, we end up with something like `phdi-k8s-record-linkage-service`. Definitely open to other naming suggestions, but it's required so we need some kind of name here. I think this is also how we'll reference the service elsewhere (i.e., in the pipeline) so having the name be consistent is probably preferable.

![Screenshot 2023-07-31 at 4 14 56 PM](https://github.com/CDCgov/phdi-azure/assets/80282552/b28f3a87-6ba0-42f3-8242-a12e39f53a26)
![Screenshot 2023-07-31 at 4 15 08 PM](https://github.com/CDCgov/phdi-azure/assets/80282552/13a405da-b42e-456b-8e65-ff6a96e4173f)

Reply:
This naming convention makes sense to me, but I don't see how we get "phdi-k8s-record-linkage-service" from `"phdi-${terraform.workspace}"`. I would think that this simply results in "phdi-k8s". Should this be something `"phdi-${terraform.workspace}-record-linkage-service"`?

Reply:
It's definitely a little opaque, but the name comes from the chart itself - see [service.yaml](https://github.com/CDCgov/phdi-charts/blob/main/charts/record-linkage/templates/service.yaml#L4). `record-linkage-service` will be appended onto whatever we declare as the release name.